src/dsu.rs

```diff
@@ -1 +1,89 @@
+// Implement (union by size) + (path compression)
+// Reference:
+// Zvi Galil and Giuseppe F. Italiano,
+// Data structures and algorithms for disjoint set union problems
```

Comment:
These comments look useful for users. Could you use `///` instead of `//` to generate documents from the comments?
Ref: https://doc.rust-lang.org/reference/comments.html#doc-comments